alert-database/src/main/java/com/synopsys/integration/alert/database/api/StaticJobAccessorV2.java

```diff
@@ -0,0 +1,225 @@
+        return distributionJobRepository.findByName(jobName)
+                   .map(this::convertToDistributionJobModel);
+    }
+
+    @Override
+    public List<DistributionJobModel> getMatchingEnabledJobs(FrequencyType frequency, Long providerConfigId, NotificationType notificationType) {
+        // TODO change this to return a page of jobs
+        return getMatchingEnabledJobs(() -> distributionJobRepository.findMatchingEnabledJob(frequency.name(), providerConfigId, notificationType.name()));
+    }
+
+    @Override
+    public List<DistributionJobModel> getMatchingEnabledJobs(Long providerConfigId, NotificationType notificationType) {
+        // TODO change this to return a page of jobs
+        return getMatchingEnabledJobs(() -> distributionJobRepository.findMatchingEnabledJob(providerConfigId, notificationType.name()));
+    }
+
+    private List<DistributionJobModel> getMatchingEnabledJobs(Supplier<List<DistributionJobEntity>> getJobs) {
+        // TODO change this to return a page of jobs
+        List<DistributionJobEntity> matchingEnabledJob = getJobs.get();
+        return matchingEnabledJob
+                   .stream()
+                   .map(this::convertToDistributionJobModel)
+                   .collect(Collectors.toList());
+    }
+
+    @Override
+    @Transactional
+    public DistributionJobModel createJob(Collection<String> descriptorNames, Collection<ConfigurationFieldModel> configuredFields) {
+        return createJobWithId(null, configuredFields, DateUtils.createCurrentDateTimestamp(), null);
+    }
+
+    @Override
+    @Transactional
+    public DistributionJobModel updateJob(UUID jobId, Collection<String> descriptorNames, Collection<ConfigurationFieldModel> configuredFields) throws AlertDatabaseConstraintException {
+        DistributionJobEntity jobEntity = distributionJobRepository.findById(jobId)
+                                              .orElseThrow(() -> new AlertDatabaseConstraintException(String.format("No job exists with the id [%s]", jobId.toString())));
+        OffsetDateTime createdAt = jobEntity.getCreatedAt();
+
+        deleteJob(jobId);
+        return createJobWithId(jobId, configuredFields, createdAt, DateUtils.createCurrentDateTimestamp());
+    }
+
+    @Override
+    @Transactional
+    public void deleteJob(UUID jobId) {
+        distributionJobRepository.deleteById(jobId);
+    }
+
+    private DistributionJobModel convertToDistributionJobModel(DistributionJobEntity jobEntity) {
+        ConfigurationJobModel configurationJobModel = convertToConfigurationJobModel(jobEntity);
+        Map<String, ConfigurationFieldModel> fields = configurationJobModel.getFieldUtility().getFields();
+        return JobConfigurationModelFieldExtractorUtils.convertToDistributionJobModel(configurationJobModel.getJobId(), fields, jobEntity.getCreatedAt(), jobEntity.getLastUpdated());
+    }
+
+    private ConfigurationJobModel convertToConfigurationJobModel(DistributionJobEntity jobEntity) {
```

Comment:
This is only used in the `convertToDistributionJobModel(DistributionJobEntity jobEntity)` method, and it looks like the only pieces we use from ConfigurationJobModel are the Job Id and the fields map. So why not change this method to just return the fields map, since the Job Id can be taken from the DistributionJobEntity?

Reply:
This method is now removed.